docs/DdnRelocatableShaderElf.md

```diff
@@ -0,0 +1,205 @@
+differences are best described with examples.  Suppose you want to load the
+descriptor for descriptor set 2 at binding 1.
+
+1. Suppose the descriptor set is not a top level user data node.  Then the
+   standard compile will add a parameter to the shader for the top level user data node
+   that contains descriptor set 2.  The corresponding `SPI_SHADER_USER_DATA`
+   entry in the PAL meta data will contain the offset of that top level node.
+   An unlinked shader may not have that offset available.  Instead the PAL metadata entry
+   will contain the number of the descriptor set that the parameter is suppose
+   to point to.  In this case, it will be 2.  In
+   `PalMetadata::fixUpRegisters` in
+   [PalMetadata.cpp](https://github.com/GPUOpen-Drivers/llpc/blob/dev/lgc/state/PalMetadata.cpp)
+   the offset is filled in.
+
+1. Suppose the descriptor set is in a top level user data node.  Then the
+   standard compile could get the descritor directly from an input register.  When
+   building an unlinked shader, this is not possible.  There are not enough
+   input register to have a 
+   separate input register for every descriptor that is potentially in a top
+   level node.  So for unlinked shaders, this is handled by spilling the user
+   data node, and indirectly loading the descriptor from the spill table.  This
+   is done in `DescBuilder::CreateLoadBufferDesc` in 
+   [DescBuilder.cpp](https://github.com/GPUOpen-Drivers/llpc/blob/dev/lgc/builder/DescBuilder.cpp).
+
+1. The final difficulty is that an unlinked shader cannot know the difference
+   between the two previous cases.  For cases where the descriptor is possibly
+   in a top level user data node, the compiler will generate code that will
+   select either the spill table or the entry for that descriptor set based on a
+   relocation. See `DescBuilder::getDescPtr` in
+   [DescBuilder.cpp](https://github.com/GPUOpen-Drivers/llpc/blob/dev/lgc/builder/DescBuilder.cpp).
+
+## Compact buffer descriptors
+
+While building an unlinked shader, it is not possible to know if the compact
+buffer descriptors will be used or not.  It is unclear if it would be worth
+generating code general enough to work in both cases.  The extra overhead would
+potentially negate any gains from using compact buffer descriptors.  So we
+decided to compile unlinked shader assuming that there are no compact buffer 
+descriptors.  If the XGL option `EnableRelocatableShaders` is true, then the
+driver will not use compact buffer descriptors.
+
+# Performance considerations
+
+Naturally, using relocatable shaders will reduce the quality of the code that is
+generated because the code has to be more general.  It is possible that this
+will affect performance, but we do not expect it to be dibilitating. This should
+be taken into considerantion when deciding to use relocatable shaders.
+
+Here is a list of known issues that could slow down relocatable shaders:
```

Comment:
Nit: are these 'issues'? I'd call them something like 'constructs' to not make it sound like these are broken and need to be somehow fixed